### PR TITLE
fix: databases recreated when delete during provisioning

### PR DESCRIPTION
**Pull request by joshuajerin**

### CODE REVIEW

File: .github/workflows/cargo-test.yaml [hunk 1]
```diff
@@ -44,6 +44,7 @@ jobs:
         run: |
           set -xe
           sudo apt-get update
+          cargo install sqlx-cli
```
Comment (sjmiller609) [inline .github/workflows/cargo-test.yaml:47]:
```suggestion
```

File: conductor/migrations/20240321181458_create_deleted_instances_table.up.sql [hunk 2]
```diff
@@ -0,0 +1,8 @@
+-- Up migration
+CREATE TABLE deleted_instances (
+    namespace VARCHAR(255) NOT NULL,
+    deleted_at TIMESTAMP WITH TIME ZONE DEFAULT CURRENT_TIMESTAMP,
+    PRIMARY KEY (namespace)
+);
+
+CREATE UNIQUE INDEX idx_deleted_instances_namespace ON deleted_instances(namespace);
```
Comment (ChuckHend) [inline conductor/migrations/20240321181458_create_deleted_instances_table.up.sql:8]: Is this writing to the queue instance or the control-plane instance?

If this is the queue instance, then we should make sure that the queue instance is dedicated to the one cluster. Up to this point the control-plane's postgres instance has been the sole location for any state related to a data-plane.

If this is the control-plane's DB, then note that this is the first introduction of the pattern of data-plane writing directly back to the control-plane database.

cc @sjmiller609 , @ianstanton

**Comment by sjmiller609** [comment]
needs SQLX_OFFLINE in the dockerfile

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -44,12 +44,41 @@ jobs:
       - self-hosted
       - dind
       - large-8x8
+    env:
```
Comment (sjmiller609) [inline .github/workflows/deploy.yml:47]: I think this file should not have any changes

File: conductor/Dockerfile [hunk 4]
```diff
@@ -23,3 +25,4 @@ RUN set -eux; \
     apt-get clean; \
     rm -rf /var/lib/apt/lists/* /var/cache/apt/archives/*
 COPY --from=builder /usr/local/cargo/bin/* /usr/local/bin/
+COPY --from=builder /build/migrations /migrations
```
Comment (sjmiller609) [inline conductor/Dockerfile:28]: I think this line is not necessary

File: conductor/migrations/20240321181458_create_deleted_instances_table.up.sql (hunk 2 above)
Comment (sjmiller609) [reply conductor/migrations/20240321181458_create_deleted_instances_table.up.sql:8]: We discussed in Slack. It is the queue DB, not the control plane DB, and I think we are both OK with this approach to save a table of deleted instances in the queue db.

**Review by sjmiller609 — CHANGES_REQUESTED** [review]

File: conductor/src/main.rs [hunk 5]
```diff
@@ -96,6 +113,43 @@ async fn run(metrics: CustomMetrics) -> Result<(), ConductorError> {
...
+            .await
+            {
+                Ok(Some(_)) => true,
+                Ok(None) => false,
+                Err(e) => {
+                    error!("Database query error: {}", e);
+                    continue;
+                }
+            };
+            println!("{}", namespace_exists);
```
Comment (sjmiller609) [inline conductor/src/main.rs:137]: remove this print statement

File: conductor/src/main.rs [hunk 6]
```diff
@@ -96,6 +113,43 @@ async fn run(metrics: CustomMetrics) -> Result<(), ConductorError> {
...
         let org_id = read_msg.message.org_id.clone();
         let instance_id = read_msg.message.inst_id.clone();
+        let namespace = format!(
+            "org-{}-inst-{}",
+            read_msg.message.organization_name, read_msg.message.dbname
+        );
+        info!("{}: Using namespace {}", read_msg.msg_id, &namespace);
+
+        if read_msg.message.event_type != Event::Delete {
+            let namespace_exists = match sqlx::query!(
```
Comment (sjmiller609) [inline conductor/src/main.rs:123]: rename variable to "namespace_already_deleted"

**Review by sjmiller609 — APPROVED** [review]

**Pull request merged**